pkg/server/api.go

```diff
@@ -80,8 +80,14 @@ func NewServerAPIHandler(s Server) *APIHandler {
 // ServeHTTP handles the requests for the machine config server
 // API handler.
 func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet && r.Method != http.MethodHead {
+		w.Header().Set("Content-Length", "0")
```

Comment:
Can you point to to a reason why we need to set this?

Reply:
> Can you point to to a reason why we need to set this?

If people start `HEAD`ing this server, it seems like they might be interested in getting valid `Content-Length` back.  I just started setting it everywhere to make it easier to test that we get it right when we actually care about it.

Reply:
So looking at https://github.com/openshift/machine-config-operator/pull/246/commits/5f0a6a7f5d29f9d97608ff325e52f62308165c5b
```
The defaultHandler gives us our usual empty 404s for requests that
don't match our other handlers.  Without it, we'd have returned Go's
default:

  404 page not found

body, and had unit-test failures like:

  --- FAIL: TestHandler (0.00s)
      --- FAIL: TestHandler/GET_/does-not-exist (0.00s)
          api_test.go:93: response body length 19 does not match Content-Length -1
```

I makes me go back to to the original thing that we should not setting/expecting `Content-Length` when **we** only care about the StatusCode.

So maybe can we remove all the `w.Header().Set("Content-Length", "0")` and only set it when we care, ie HEAD/GET success result.

Reply:
> It makes me go back to to the original thing that we should not setting/expecting `Content-Length` when *we* only care about the StatusCode.

If we want to return `404 page not found\n` (with `Content-Type: text/plain; charset=utf-8`?) for all of our 404s, I'm ok with that (although I don't see a need to do it).  But I would like to be consistent among our 404s, and not have some that return with no body and some that return with the text/plain body.

Reply:
lets keep it as is for now..